presto-hive/src/main/java/com/facebook/presto/hive/HiveSessionProperties.java

```diff
@@ -670,7 +671,12 @@ public HiveSessionProperties(HiveClientConfig hiveClientConfig, OrcFileWriterCon
                         HUDI_METADATA_ENABLED,
                         "For Hudi tables prefer to fetch the list of file names, sizes and other metadata from the internal metadata table rather than storage",
                         hiveClientConfig.isHudiMetadataEnabled(),
-                        false));
+                        false),
+                booleanProperty(
+                        PROVIDE_EXISTING_PARTITION_VERSION_FOR_OVERWRITE,
+                        "Use partition version for overwriting an existing partition in order to support optimistic concurrency control",
+                        hiveClientConfig.isProvideExistingPartitionVersionForOverwrite(),
+                        true));
```

Comment:
Partition version is not there in oss HMS. So how about we make it a prism session property and use it in PrismPartitionObjectBuilder to decide whether or not to set the partition version in the new parition object

Reply:
I was initially thinking of letting it available in OSS side so that any other non facebook HMS extension can use it with minimal changes. But I think it makes sense not to do this right now provided there may not be a need for this on OSS side. I moved the session related changes out to prism.